biomage/account/account.py

```diff
@@ -0,0 +1,202 @@
+
+
+@click.group()
+def account():
+    """
+    Manage  Cellscope account information.
+    """
+    pass
+
+
+COGNITO_PRODUCTION_POOL = os.getenv("COGNITO_PRODUCTION_POOL")
+COGNITO_STAGING_POOL = os.getenv("COGNITO_STAGING_POOL")
+
+
+def generate_password():
+    today = time.strftime("%Y-%m-%d")
+    return (
+        "Tmp_"
+        + today
+        + "".join([choice(string.ascii_uppercase + string.digits) for _ in range(8)])
+    )
+
+
+def create_account(username, email, userpool):
+    """
+    Creates a new account with the information provided. Requires a password change call afterwards."""
+    p = Popen(
+        f"""aws cognito-idp admin-create-user \
+            --user-pool-id "{userpool}" \
+            --username "{email}" \
+            --message-action "SUPPRESS" \
+            --user-attributes \
+                "Name=email,Value={email}" \
+                "Name=name,Value='{username}'" \
+                "Name=email_verified,Value=true" """.format(
+            userpool=userpool, email=email, username=username
+        ),
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        shell=True,
+    )
+
+    output, error = p.communicate(input=b"\n")
+
+    print(output)
+
+    return error
+
+
+@click.command()
+@click.option(
+    "--email",
+    required=True,
+    help="User email for the account to change the password in production.",
+)
+@click.option(
+    "--password",
+    required=True,
+    help="Password for the new account.",
+)
+def change_password(email, password, userpool):
+
+    return _change_password(email, password, userpool)
+
+
+def _change_password(email, password, userpool):
+    p = Popen(
+        f"""aws cognito-idp admin-set-user-password \
+            --user-pool-id "{userpool}" \
+            --username "{email}" \
+            --password "{password}" \
+            --permanent""".format(
+            userpool=userpool, email=email, password=password
+        ),
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        shell=True,
+    )
+    output, error = p.communicate(input=b"\n")
+    print(output)
+
+    return error
+
+
+@click.command()
+@click.option(
+    "--email",
+    required=True,
+    help="User email for the account.",
+)
+@click.option(
+    "--username",
+    required=True,
+    help="The first and last name for the user. (e.g.: Arthur Dent)",
+)
+@click.option(
+    "--password",
+    required=False,
+    help="Password for the new account.",
+)
+def create_user(username, email, password, userpool):
+    """
+    Creates a new account with the provided password. The user will not receive any
+    email and the account & email will be marked as verified.
+    """
+    if not password:
+        password = generate_password()
+
+    print("%s,%s,%s" % (username, email, password))
+    _create_user(username, email, password, userpool)
+
+
+def _create_user(username, email, password, userpool, overwrite=False):
```

Comment:
the term `username` in Cognito has a special meaning, it's the UUID assigned by Cognito to the user. Would it be better to rename the variable `username` into `user_name` to make it easier to distinguish it from Cognito's `username`?